tests/test_list.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any
+
+from .mockers import (GitRepositorySandbox, assert_command, launch_command, mock_run_cmd)
+
+
+class TestList:
+
+    def setup_method(self) -> None:
+        self.repo_sandbox = GitRepositorySandbox()
+
+        (
+            self.repo_sandbox
+            # Create the remote and sandbox repos, chdir into sandbox repo
+            .new_repo(self.repo_sandbox.remote_path, "--bare")
+            .new_repo(self.repo_sandbox.local_path)
+            .execute(f"git remote add origin {self.repo_sandbox.remote_path}")
+            .execute('git config user.email "tester@test.com"')
+            .execute('git config user.name "Tester Test"')
+        )
+
+    def test_list(self, mocker: Any) -> None:
+        """
+        Verify behaviour of a 'git machete list' command.
+        """
+        mocker.patch('git_machete.utils.run_cmd', mock_run_cmd)  # to hide git outputs in tests
+
+        (
+            self.repo_sandbox.new_branch("master")
+                .commit("master commit.")
+                .new_branch("develop")
+                .commit("develop commit.")
+                .new_branch("feature_0")
+                .commit("feature_0 commit.")
+                .new_branch("feature_0_0")
+                .commit("feature_0_0 commit.")
+                .new_branch("feature_0_0_0")
+                .commit("feature_0_0_0 commit.")
+                .check_out("feature_0")
+                .new_branch("feature_0_1")
+                .commit("feature_0_1 commit.")
+                .check_out("develop")
+                .new_branch("feature_1")
+                .commit("feature_1 commit.")
+        )
+        launch_command("discover", "-y")
+
+        (
+            self.repo_sandbox.check_out("develop")
+                             .new_branch("feature_2")
+                             .commit("feature_2 commit.")
+        )
+
+        expected_msg = """
```

Comment:
Here and in other places
```suggestion
        expected_output = """
```